OSDOCS-11255 OCP 4.16.2 Release Notes (#78608)
openshift/openshift-docs: release_notes/ocp-4-16-release-notes.adoc

* Previously, unexpected output would appear in the terminal when creating an installer-provisioned infrastructure (IPI) cluster. With this release, the issue has been resolved and the unexpected output no longer appears. (link:https://issues.redhat.com/browse/OCPBUGS-36156[*OCPBUGS-36156*])

* Previously, the {product-title} console did not show filesystem metrics on the nodes list. With this release, the filesystem metrics now appear in the nodes table. (link:https://issues.redhat.com/browse/OCPBUGS-35946[*OCPBUGS-35946*])

* Previously, the Prometheus dashboard showed up empty for non-multi-cluster environments. With this release, the dashboard populates the dashboard panels as expected for both cases. (link:https://issues.redhat.com/browse/OCPBUGS-35904[*OCPBUGS-35904*])

* Previously, a regression in 4.16.0 caused new baremetal installer-provisioned infrastructure (IPI) installations to fail when proxies were used. This was caused by one of the services in the bootstrap virtual machine (VM) trying to access IP address 0.0.0.0 through the proxy. With this release, this service no longer accesses 0.0.0.0. (link:https://issues.redhat.com/browse/OCPBUGS-35818[*OCPBUGS-35818*])

> * Previously, the current IBM Power Virtual Server (Power VS) CAPI provider waits for some resources to be created before creating the load balancers, it is possible that the load balancers will not finish before the 15 minute timeout. With this release, the timeout has been increased. (link:https://issues.redhat.com/browse/OCPBUGS-35722[*OCPBUGS-35722*])

Reviewer: ```suggestion
* Previously, the current IBM Power Virtual Server (Power VS) Cluster API provider waited for some resources to be created before creating the load balancers, which sometimes resulted in the load balancers not being created before the 15 minute timeout. With this release, the timeout has been increased. (link:https://issues.redhat.com/browse/OCPBUGS-35722[*OCPBUGS-35722*])
```

> * Previously, the current IBM Power Virtual Server (Power VS) Cluster API provider waited for some resources to be created before creating the load balancers, which sometimes resulted in the load balancers not being created before the 15 minute timeout. With this release, the timeout has been increased. (link:https://issues.redhat.com/browse/OCPBUGS-35722[*OCPBUGS-35722*])

Reviewer 2: `{cap-ibm-first}` resolves to "Cluster API Provider IBM Cloud", which is a different name but apparently they are using [the same Cluster API Provider](https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud) for Power VS. Added `{ibm-power-server-title}` ("IBM Power Virtual Server") to try to keep it clear that Power VS was impacted as per original text.

```suggestion
* Previously, the {cap-ibm-first} waited for some resources to be created before creating the load balancers on {ibm-power-server-title} clusters. This delay sometimes resulted in the load balancers not being created before the 15 minute timeout. With this release, the timeout has been increased. (link:https://issues.redhat.com/browse/OCPBUGS-35722[*OCPBUGS-35722*])
```

> * Previously, in CAPI-based installers on the OpenStack platform, the additional security group rule added to master nodes for compact clusters was forcing IPv4 protocol and prevented deploying dual-stack clusters. This was a regression from the terraform-based installer. With this release, the rule now uses the correct protocol based on the requested IP version. (link:https://issues.redhat.com/browse/OCPBUGS-35718[*OCPBUGS-35718*])

Reviewer: ```suggestion
* Previously, when installing a cluster on :rh-openstack-first: using the Cluster API implementation, the additional security group rule added to master nodes for compact clusters was forcing IPv4 protocol and prevented deploying dual-stack clusters. This was a regression from installations using Terraform. With this release, the rule now uses the correct protocol based on the requested IP version. (link:https://issues.redhat.com/browse/OCPBUGS-35718[*OCPBUGS-35718*])
```

> * Previously, when installing a cluster on :rh-openstack-first: using the Cluster API implementation, the additional security group rule added to master nodes for compact clusters was forcing IPv4 protocol and prevented deploying dual-stack clusters. This was a regression from installations using Terraform. With this release, the rule now uses the correct protocol based on the requested IP version. (link:https://issues.redhat.com/browse/OCPBUGS-35718[*OCPBUGS-35718*])

Reviewer 2: ```suggestion
* Previously, when installing a cluster on {rh-openstack-first} using the Cluster API implementation, the additional security group rule added to control plane nodes for compact clusters was forcing IPv4 protocol and prevented deploying dual-stack clusters. This was a regression from installations using Terraform. With this release, the rule now uses the correct protocol based on the requested IP version. (link:https://issues.redhat.com/browse/OCPBUGS-35718[*OCPBUGS-35718*])
```

* Previously, the internal image registry would not correctly authenticate users on clusters configured with external OpenID Connect (OIDC) users, making it impossible for users to push or pull images to and from the internal image registry. With this release, the internal image registry starts using the SelfSubjectReview API, dropping use of the {product-title} specific user API, which is not available on clusters configured with external OIDC users, making it possible to successfully authenticate with the image registry again. (link:https://issues.redhat.com/browse/OCPBUGS-35567[*OCPBUGS-35567*])

> * Prevously, an errant code change resulted in a duplicated `oauth.config.openshift.io` item on the Global Configuration page. With this update, the duplicated item is removed. (link:https://issues.redhat.com/browse/OCPBUGS-35565[*OCPBUGS-35565*])

Reviewer 2: ```suggestion
* Prevously, an errant code change resulted in a duplicated `oauth.config.openshift.io` item on the *Global Configuration* page. With this update, the duplicated item is removed. (link:https://issues.redhat.com/browse/OCPBUGS-35565[*OCPBUGS-35565*])
```

> * Previously, with `oc-mirror` v2 Tech Preview, when mirroring fails due to various reasons (network, invalid operator catalog content, etc), `oc-mirror` does not generate cluster resources. With this bug fix, `oc-mirror` v2 Tech Preview:

Reviewer: ```suggestion
* Previously, with `oc-mirror` v2, when mirroring fails due to various reasons, such as network errors or invalid operator catalog content, `oc-mirror` did not generate cluster resources. With this bug fix, `oc-mirror` v2:
```

> * Previously, with `oc-mirror` v2, when mirroring fails due to various reasons, such as network errors or invalid operator catalog content, `oc-mirror` did not generate cluster resources. With this bug fix, `oc-mirror` v2:

Reviewer 2: ```suggestion
* Previously, with `oc-mirror` v2, when mirroring fails due to various reasons, such as network errors or invalid operator catalog content, `oc-mirror` did not generate cluster resources. With this bug fix, `oc-mirror` v2 performs the following actions:
```

>  ** Pursues mirroring other images when errors occur on operator images and additional images, and aborts mirroring when errors occur on release images.

Reviewer: ```suggestion
 ** Pursues mirroring other images when errors occur on Operator images and additional images, and aborts mirroring when errors occur on release images.
```

 ** Generates cluster resources for the cluster based on subset of correctly mirrored images.
 ** Collects all mirroring errors in a log file.

>  ** logs all mirroring errors in a separate log file.

Reviewer: ```suggestion
 ** Logs all mirroring errors in a separate log file.
```

* Previously, builds could not set the `GIT_LFS_SKIP_SMUDGE` environment variable and use its value when cloning source code. This caused builds to fail for some git repositories with LFS files. With this release, the build is allowed to set this environment variable and use it during the git clone step of the build. (link:https://issues.redhat.com/browse/OCPBUGS-35283[*OCPBUGS-35283*])

* Previously, registry overrides were present in non-relevant data plane images. With this release, the way {product-title} propagates the override-registries has been modified and the issue is fixed. (link:https://issues.redhat.com/browse/OCPBUGS-34602[*OCPBUGS-34602*])

* Previously, RegistryMirrorProvider images were not being updated during the reconciliation because RegistryMirrorProvider was modifying the cached image directly instead of the internal entries. With this release, the way we update the images has been modified, avoiding the cache and doing it directly in the entries so the bug no longer presents. (link:https://issues.redhat.com/browse/OCPBUGS-34569[*OCPBUGS-34569*])

> * Previously, the `alertmanager-trusted-ca-bundle` configmap was not injected into the user-defined Alertmanager container, which prevented the verification of HTTPS web servers receiving alert notifications. With this update, the trusted CA bundle configmap is mounted into the Alertmanager container at the `/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem` path. (link:https://issues.redhat.com/browse/OCPBUGS-34530[*OCPBUGS-34530*])

Reviewer: ```suggestion
* Previously, the `alertmanager-trusted-ca-bundle` `ConfigMap` was not injected into the user-defined Alertmanager container, which prevented the verification of HTTPS web servers receiving alert notifications. With this update, the trusted CA bundle `ConfigMap` is mounted into the Alertmanager container at the `/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem` path. (link:https://issues.redhat.com/browse/OCPBUGS-34530[*OCPBUGS-34530*])
```

> * Previously, in Amazon Web Services (AWS) security token service (STS), Cloud Credential Operator (CCO) checked `awsSTSIAMRoleARN` in `CredentialsRequest` to create the secret. When awsSTSIAMRoleARN was not present, CCO logged the error. With this release, CCO no longer logs the error and the bug is fixed. (link:https://issues.redhat.com/browse/OCPBUGS-34117[*OCPBUGS-34117*])

Reviewer: ```suggestion
* Previously, in Amazon Web Services (AWS) security token service (STS), Cloud Credential Operator (CCO) checked `awsSTSIAMRoleARN` in `CredentialsRequest` to create the secret. When `awsSTSIAMRoleARN` was not present, CCO logged the error. With this release, CCO no longer logs the error and the bug is fixed. (link:https://issues.redhat.com/browse/OCPBUGS-34117[*OCPBUGS-34117*])
```

> * Previously, in Amazon Web Services (AWS) security token service (STS), Cloud Credential Operator (CCO) checked `awsSTSIAMRoleARN` in `CredentialsRequest` to create the secret. When `awsSTSIAMRoleARN` was not present, CCO logged the error. With this release, CCO no longer logs the error and the bug is fixed. (link:https://issues.redhat.com/browse/OCPBUGS-34117[*OCPBUGS-34117*])

Reviewer 2: ```suggestion
* Previously, for {aws-first} clusters that use {sts-first}, the Cloud Credential Operator (CCO) checked the value of `awsSTSIAMRoleARN` in  the `CredentialsRequest` custom resource to create a secret. When `awsSTSIAMRoleARN` was not present, CCO logged an error. The issue is resolved in this release. (link:https://issues.redhat.com/browse/OCPBUGS-34117[*OCPBUGS-34117*])
```

* Previously, with the OVN-Kubernetes setting for routing-via-host set to shared gateway mode, its default value, OVN-Kubernetes did not correctly handle traffic streams that mixed non-fragmented and fragmented packets from the IP layer on cluster ingress. This caused connection resets or packet drops. With this release, OVN-Kubernetes correctly reassembles and handles external traffic IP packet fragments on ingress. (link:https://issues.redhat.com/browse/OCPBUGS-29511[*OCPBUGS-29511*])

[id="ocp-4-16-2-known-issue_{context}"]
==== Known issue

> * If the `ConfigMap` maximum transmission unit (MTU) is absent in the namespace 'openshift-network-operator', users have to create the `ConfigMap` manually with the machine MTU value, before starting the live migration. Otherwise, the live migration will get stuck and fail. (link:https://issues.redhat.com/browse/OCPBUGS-35829[*OCPBUGS-35829*])

Reviewer: ```suggestion
* If the `ConfigMap` maximum transmission unit (MTU) is absent in the namespace `openshift-network-operator`, users have to create the `ConfigMap` manually with the machine MTU value, before starting the live migration. Otherwise, the live migration will get stuck and fail. (link:https://issues.redhat.com/browse/OCPBUGS-35829[*OCPBUGS-35829*])
```
